server/api/accounts.js

```diff
@@ -16,15 +16,24 @@ const register = function (server, serverOptions) {
         path: '/api/accounts',
         options: {
             tags: ['api','accounts'],
+            description: 'Get a paginated list of all accounts (need admin role)',
+            notes: 'This endpoint returns a paginated list of accounts that have been created. This is a much longer detailed description if we have answers to common questions, but I mainly have questions. Whats the difference between accounts and users? When would you want to list all accounts instead of all users? Assuming you need an admin role to use this endpoint? Where is that configured?',
             auth: {
                 scope: 'admin'
             },
             validate: {
                 query: {
-                    sort: Joi.string().default('_id'),
-                    limit: Joi.number().default(20),
-                    page: Joi.number().default(1)
+                    sort: Joi.string().default('_id').description('This is a description for this query parameter. This is obviously a sort but what are the allowed values in the default setup? I\'ve added an example allow() validation which will show up as a pick list in interactive docs.').allow('_id', '_option2', '_option3'),
+                    limit: Joi.number().integer().default(20).description('I\'m assuming this is how many results you want at a time per page? We should mention that just so it\'s clear. It should also have min/max Joi validation which also shows up in docs. I\'ve added an example.').min(1).max(1000),
+                    page: Joi.number().integer().default(1).description('Looks like default is 1 so this is a 1 based index not a zero based index right? We should state stuff like that here.').min(1).max(1000)
                 }
+            },
+            response: {
+              schema: Joi.object({
+                results: Joi.array().items(
+                  Account.schema
+                ).label('Array of Account objects').required()
+              }).label('Accounts List Response Model').description('Returns a list of accounts.')
```

Comment:
It may be worth it to review these response objects, they're not always clean versions of the model schemas. Sometimes you customize the response based on user permissions. In that case the response object isn't the same in all cases.

Reply:
Ok, there are different ways we can handle that, Joi is pretty flexible. By default it makes every object and property optional and you have to add a .required() to make it validation enforced so if a field isn't always returned we could make it optional to the validator. You can also declare validation patterns, like when response object keys are dynamic. You can also add a "sample: 0" to the route to have the response validation completely turned off and only there for reference.